src/databricks/labs/mcp/servers/unity_catalog/server.py

```diff
@@ -44,23 +52,105 @@ def get_tools_dict(settings) -> dict[str, BaseTool]:
     }
 
 
-async def start() -> None:
-    server = Server(name="mcp-unitycatalog", version=VERSION)
+def start() -> None:
+    # server = Server(name="mcp-unitycatalog", version=VERSION)
+    server = FastMCP(name="mcp-unitycatalog", version=VERSION)
     tools_dict = get_tools_dict(settings=get_settings())
 
-    @server.list_tools()
-    async def list_tools() -> list[ToolSpec]:
-        return [tool.tool_spec for tool in tools_dict.values()]
 
-    @server.call_tool()
-    async def call_tool(name: str, arguments: dict) -> list[Content]:
-        tool = tools_dict[name]
-        return tool.execute(**arguments)
+    
+    @server.tool()
+    def run_sql(sql: str) -> str:
+        """
+        Run a stateless, read-only SQL query in Databricks.
+        
+        Can be used to also query metadata, e.g. 
+        1. To discover tables and other data sources
+        via commands like `SHOW CATALOGS`, `SHOW SCHEMAS in <catalog_name>`, `SHOW TABLES IN <catalog_name>.<schema_name>`, etc.
+        2. To understand the schema of a table, e.g. `DESCRIBE TABLE <table>` or `DESCRIBE TABLE EXTENDED <table>` to enable generating subsequent SQL queries.
+
+
+        Note that each command execution is a separate Databricks SQL statement, and that the session state is not retained between command executions.
+
+        :param sql: The SQL query to run
+        :return: The result of the SQL query
+        """
+        w = WorkspaceClient()
+        warehouse_id = w.warehouses.list()[0].id
+        return w.statement_execution.execute_statement(sql, warehouse_id)    
+    
+
+    @server.tool()
+    def create_genie_space(name: str, table_full_names: list[str]) -> str:
+        """ 
+        Create a new genie space.
```

Comment:
I was trying this out in combination with the search tool & `ask_genie` tool (before I added the SQL tool) to enable agents to create spaces and talk to them. It seemed to work pretty well; we can add a similar tool under our genie spaces MCP server (`/api/2.0/mcp/genie/...`)